nncf/common/graph/layer_attributes.py

```diff
@@ -123,8 +127,11 @@ def get_weight_shape(self) -> List[int]:
             return [self.out_channels, self.in_channels // self.groups, *self.kernel_size]
         return [self.in_channels, self.out_channels // self.groups, *self.kernel_size]
 
-    def get_num_filters(self) -> int:
-        return self.out_channels
+    def get_target_dim_for_compression(self) -> int:
+        # Always quantize per each "out" channel
+        if self.transpose:
+            return 1
+        return 0
```

Comment:
It looks like this is not applicable for TF models.

Reply:
@alexsu52 how do you propose to resolve the fact that some common algos require exact definitions for what each weight tensor dimension means, and that at the same time the layouts of weight tensors are different between the backends?

Reply:
I guess, the backend should provide this information.

Reply:
Yes, and the backend should do so at the backend level of understanding, not at the layer attribute level which is common.

Reply:
I will not perpetuate the `__nncf_backend__` in any further PRs, and because proper resolution of this requires a separate effort, we will resolve this in a proper way once we get to integrating the quantization algo code with TF. Right now I need this fix to get correct performance with NNCF-produced PT UNet models.

Reply:
Created ticked 58678 to track the upcoming tensor layout backend abstraction.